src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/RotatingStar.cpp

```diff
@@ -232,12 +231,27 @@ std::array<double, 6> CstSolution::interpolate(
                         << num_radial_points_);
   }
 
-  const size_t radial_stencil_index = static_cast<size_t>(std::clamp(
-      static_cast<int>(radius_index) - static_cast<int>(stencil_size) / 2, 0,
-      static_cast<int>(num_radial_points_ - stencil_size)));
+  // If the stencil size is an even number, a shift in the stencil may be
+  // necessary. The intention is that the target point is always in the middle
+  // of the stencil, i.e. between the middle 2 points.
+  int radial_stencil_adjust = 0;
+  int angular_stencil_adjust = 0;
+  if constexpr (stencil_size % 2 == 0) {
+    if (target_radius > radius_[radius_index * num_angular_points_]) {
+      radial_stencil_adjust++;
+    };
+    if (target_abs_cos_theta > cos_theta_[angular_index]) {
+      angular_stencil_adjust++;
+    };
```

Comment:
[optional] Two extra semicolons.

Reply:
Squashed the semicolons immediately. Added the unit test in a fixup commit. The test attempts to interpolate at a point which would've caused the original stencil construction to fail (I tested it with the old method and it did indeed fail).